pkg/schemapatcher/gen.go

```diff
@@ -330,24 +341,26 @@ func crdsFromDirectory(ctx *genall.GenerationContext, dir string) (map[schema.Gr
 		if err := kyaml.Unmarshal(rawContent, &typeMeta); err != nil {
 			continue
 		}
-		if typeMeta.APIVersion != apiextAPIVersion || typeMeta.Kind != "CustomResourceDefinition" {
+		if typeMeta.Kind != "CustomResourceDefinition" {
 			continue
 		}
 
-		// collect the group-kind and versions from the actual structured form
-		var actualCRD apiext.CustomResourceDefinition
-		if err := kyaml.Unmarshal(rawContent, &actualCRD); err != nil {
-			continue
-		}
-		groupKind := schema.GroupKind{Group: actualCRD.Spec.Group, Kind: actualCRD.Spec.Names.Kind}
+		writeGlobalSchema := false
+		var groupKind *schema.GroupKind
 		var versions map[string]struct{}
-		if len(actualCRD.Spec.Versions) == 0 {
-			versions = map[string]struct{}{actualCRD.Spec.Version: struct{}{}}
+		// collect the group-kind and versions from the actual structured form
+		if typeMeta.APIVersion == apiextv1APIVersion {
+			groupKind, versions = unmarshalCRDV1(rawContent)
+			// v1 CRD does not have a global schema
+		} else if typeMeta.APIVersion == apiextv1b1APIVersion {
+			groupKind, versions = unmarshalCRDV1beta1(rawContent)
+			// v1beta1 CRD has a global schema
+			writeGlobalSchema = true
 		} else {
-			versions = make(map[string]struct{}, len(actualCRD.Spec.Versions))
-			for _, ver := range actualCRD.Spec.Versions {
-				versions[ver.Name] = struct{}{}
-			}
+			return nil, fmt.Errorf("Unknown CRD apiVersion %q", typeMeta.APIVersion)
+		}
+		if groupKind == nil {
+			continue
```

Comment:
can this happen now?

Reply:
Yes - both unmarshal functions returns `nil, nil` if unmarshaling fails (silent fail). This was the previous behavior so I figured it should be retained.  Though maybe klog should be in play to provide output when this happens?

Reply:
Would expect a warning, yes.

Reply:
Why is there no error return value in unmarshalCRDV1/v1beta1?

Reply:
Updated to return error values and log them as warnings.